src/pq_flash_index.cpp

```diff
@@ -252,11 +252,23 @@ void PQFlashIndex<T, LabelT>::generate_cache_list_from_sample_queries(std::strin
     std::vector<uint64_t> tmp_result_ids_64(sample_num, 0);
     std::vector<float> tmp_result_dists(sample_num, 0);
 
+    bool filtered_search = false;
+    std::vector<LabelT> random_query_filters(sample_num);
+    if (_filter_to_medoid_ids.size() != 0)
+    {
+        filtered_search = true;
+        generate_random_labels(random_query_filters, sample_num, nthreads);
```

Comment:
![image](https://github.com/microsoft/DiskANN/assets/15103725/9a5c0230-db8b-4b6c-aab9-5ccd1e3614b3)

Reply:
you need to fix these warnings

Reply:
done